### ISSUE TITLE
error defining array type in UnitTest function

### ISSUE THREAD

**Issue by lucaferranti** [chapel-lang/chapel#20939]
### Summary of Problem

The following works nicely 

```chapel
type D = [0..1] int;
```

however if I wrap that definition inside a function test for UnitTest

```chapel
use UnitTest;

proc testNotWorking(test: borrowed Test) throws {
  type D = [0..1] int;
}

UnitTest.main();
```

When compiling with `chpl  tmp.chpl` I get the error

```
tmp.chpl:4: In function 'testNotWorking':
tmp.chpl:5: error: Iterators not allowed in first class functions
tmp.chpl:4: In function 'testNotWorking':
tmp.chpl:5: error: Iterators not allowed in first class functions
```

Is this expected/intended? 

### Configuration Information

- Output of `chpl --version`:
```
chpl version 1.28.0
  built with LLVM version 12.0.0
Copyright 2020-2022 Hewlett Packard Enterprise Development LP
Copyright 2004-2019 Cray Inc.
(See LICENSE file for more details)
```

- Output of `$CHPL_HOME/util/printchplenv --anonymize`:

```
CHPL_TARGET_PLATFORM: linux64
CHPL_TARGET_COMPILER: llvm
CHPL_TARGET_ARCH: x86_64
CHPL_TARGET_CPU: native
CHPL_LOCALE_MODEL: flat
CHPL_COMM: none
CHPL_TASKS: qthreads
CHPL_LAUNCHER: none
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
CHPL_ATOMICS: cstdlib
CHPL_GMP: none
CHPL_HWLOC: bundled
CHPL_RE2: none
CHPL_LLVM: system
CHPL_AUX_FILESYS: none
```

- Back-end compiler and version, e.g. `gcc --version` or `clang --version`:
```
gcc (Ubuntu 10.3.0-1ubuntu1) 10.3.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by bmcdonald3** [comment]
Thanks for pointing this out. I looked into this briefly, and it appears that this issue was fixed by https://github.com/chapel-lang/chapel/pull/20554 and you shouldn't have this problem anymore if using Chapel's `main` branch. I think what was going on was that the `UnitTest` library is trying to capture the contents of the function as a FCF and our FCF is currently a work-in-progress, so just didn't support this functionality in the last release.

Hope this wasn't too much of a blocker for you, but it should be fixed in the next release!
